broker/plugins/kubejobs/plugin.py

```diff
@@ -233,6 +235,70 @@ def errors(self):
             return ()
         return self.rds.lrange("job:errors", 0, -1)
 
+    def validate(self, data):
+        data_model = {
+            "cmd": list,
+            "config_id": six.string_types,
+            "control_parameters": dict,
+            "control_plugin": six.string_types,
+            "env_vars": dict,
+            "img": six.string_types,
+            "init_size": int,
+            "monitor_info": dict,
+            "monitor_plugin": six.string_types,
+            "enable_auth": bool,
+            # The parameters below are only needed if enable_auth is True
+            # "username": six.string_types
+            # "password": six.string_types
+            "redis_workload": six.string_types,
+            "enable_visualizer": bool
+            # The parameters below are only needed if enable_visualizer is True
+            # "visualizer_plugin": six.string_types
+            # "visualizer_info":dict
+        }
+
+        for key in data_model:
+            if (key not in data):
+                raise ex.BadRequestException(
+                    "Variable \"{}\" is missing".format(key))
+            if (not isinstance(data[key], data_model[key])):
+                raise ex.BadRequestException(
+                    "\"{}\" has unexpected variable type".format(key))
+
+        if (data["enable_visualizer"]):
+            if ("visualizer_plugin" not in data):
+                raise ex.BadRequestException(
+                    "Variable \"visualizer_plugin\" is missing")
+            if (not isinstance(data["visualizer_plugin"], six.string_types)):
+                raise ex.BadRequestException(
+                    "\"visualizer_plguin\" has unexpected variable type")
+
+            if ("visualizer_info" not in data):
+                raise ex.BadRequestException(
+                    "Variable \"visualizer_info\" is missing")
+            if (not isinstance(data["visualizer_info"], dict)):
+                raise ex.BadRequestException(
+                    "\"visualizer_info\" has unexpected variable type")
+
+        if (data["enable_auth"]):
```

Comment:
checking auth is outside of plugin scope and should be performed before its creation.